### PR TITLE
Use google-style docstrings and sphinx for doc generation

### PR DESCRIPTION
**Pull request by jmorel**
Converts all client docstrings to google-style docstrings and uses sphinx to extract them and build the references.

This will also make it easier to include in the `substra-documentation` repository.

You can browse the end result there: https://github.com/SubstraFoundation/substra/blob/sphinx_google_docstring/references/sdk.md
And compare it with the current version: https://github.com/SubstraFoundation/substra/blob/master/references/sdk.md

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+author = 'Substra Foundation'
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx.ext.napoleon'
```
Comment (natct10) [inline conf.py:31]: :+1:

File: substra/sdk/client.py [hunk 2]
```diff
@@ -371,21 +377,31 @@ def add_algo(self, data, exist_ok=False):
...
+                    }
+
+            exist_ok (boolean): If true, AlreadyExists exceptions will be ignored and the
+                existing asset will be returned.
+
+        Returns:
+            The newly created asset as a dict.
+
+        Raises:
+            AlreadyExists: a data sample with the same content already exists on the server.
```
Comment (natct10) [inline substra/sdk/client.py:404]: Awesome!

**Comment by jmorel** [comment]
@samlesu I updated all docstrings but kept sphinx because the markdown output generated by pydocmd was unreadable.

**Comment by samlesu** [comment]
Thanks @jmorel for improving the docstrings and for moving to the Google notations.

IMHO, the generated markdown documentation for the SDK is not readable. So I would prefer to remove it (I think the source code is way easier to read). I think it's ok as the SDK documentation will be available in the Substra web documentation.

**Comment by jmorel** [comment]
Since this PR makes the generated references hard to read and since the work for proper integration with substra-documentation is yet to come, I'm closing this. 

It could be reopen later and even largely re-used when substra-documentation starts depending on substra.

**Pull request closed without merging**